### PR TITLE
convert bibtexkeypatterndlg to javafx as well

### PR DESCRIPTION
**Pull request by Siedlerchr**
Followup from #4253  in the Bibtex key pattern panel there was an exception thrown if you had defined custom entry types or used biblatex mode.

@Super-Tang In your PR you used a fixed size for the number of labels and buttons. However, as it is possible in JabRef to customize the entry types, e.g. add new ones, the size is no longer fixed. Therefore if the number of entry types was > 19 the array threw an exception.
I now used the size of the entry type list to setup the array size and wrapped it in a Scroll Pane

Edit// I set colspan to remaining, so it uses all space
grafik

----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [x] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

**Comment by Siedlerchr** [comment]
I decided to quickly convert the Bibtex Key Pattern Dialog as well to javafx., it is using the same panel from the pref tabs

**Comment by Super-Tang** [comment]
Sorry for not thinking so much. :( @Siedlerchr

**Comment by Siedlerchr** [comment]
@Super-Tang  No problem! Now you know it for the future and I bet you won't come in such a situation again ;)

**Comment by tobiasdiez** [comment]
I implemented a similar fix in #4280, which goes even a step further and completely removes the textbox array. Sorry for not being quicker with my PR and the double effort. So that we don't run in a merge hell, I would propose you remove your changes to this key pattern pane class. The code looks good otherwise.

File: src/main/java/org/jabref/gui/actions/BibtexKeyPatternAction.java [hunk 1]
```diff
@@ -5,16 +5,15 @@
...
     public BibtexKeyPatternAction(JabRefFrame frame) {
         this.frame = frame;
     }
 
     @Override
     public void execute() {
-        BibtexKeyPatternDialog bibtexKeyPatternDialog = new BibtexKeyPatternDialog(frame.getCurrentBasePanel());
-        bibtexKeyPatternDialog.setLocationRelativeTo(null);
-        bibtexKeyPatternDialog.setVisible(true);
+        BibtexKeyPatternDialog dlg = new BibtexKeyPatternDialog(frame.getCurrentBasePanel());
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/actions/BibtexKeyPatternAction.java]: Please no abbreviations

File: src/main/java/org/jabref/gui/bibtexkeypattern/BibtexKeyPatternDialog.java [hunk 2]
```diff
@@ -44,53 +24,20 @@ public BibtexKeyPatternDialog(BasePanel panel) {
...
-        Action cancelAction = new AbstractAction() {
-
-            @Override
-            public void actionPerformed(ActionEvent e) {
-                dialog.dispatchEvent(new WindowEvent(dialog, WindowEvent.WINDOW_CLOSING));
+        this.setResultConverter(button -> {
+            if (button == ButtonType.APPLY) {
+                metaData.setCiteKeyPattern(bibtexKeyPatternPanel.getKeyPatternAsDatabaseBibtexKeyPattern());
+                panel.markNonUndoableBaseChanged();
+            } else {
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/bibtexkeypattern/BibtexKeyPatternDialog.java]: else branch is not really needed

File: src/main/java/org/jabref/gui/util/BaseDialog.java [hunk 3]
```diff
@@ -20,7 +20,7 @@ protected BaseDialog() {
         });
 
         setDialogIcon(IconTheme.getJabRefImageFX());
-
+        setResizable(true);
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/util/BaseDialog.java]: It's a good idea to set this globally. Can you please remove the other explicit `setResizable` calls that are now no longer needed.

Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/util/BaseDialog.java]: I removed it everywhere except in the prefs dialog to not create a conflict with your other PR

**Pull request merged**